src/primitives.jl

```diff
@@ -179,6 +181,30 @@ end
 dropnull(X::NullableVector) = X.values[!X.isnull]    # -> Vector
 
 """
+    dropnull!(X::AbstractVector)
+
+Remove null entries of `X` in-place and return the updated vector. This is a
+no-op if no nulls are present.
+
+    dropnull!(X::NullableVector)
+
+Remove null entries of `X` in-place and return a `Vector` view of `X.values`
+"""
+function dropnull!(X::AbstractVector)
+    if !(Nullable <: eltype(X))
+        return X
+    end
+    deleteat!(X, find(isnull, X))
```

Comment:
I agree that the methods could be merged, but I don't see how to get around the `isa` check when `eltype(X)` is `Any`. How could we infer which entries in a `Vector{Any}` are of a specific type (`Nullable` in this case) without a manual check?

The two methods could be merged with something like this?
```julia
function dropnull!{T}(X::AbstractVector{T})
    # if X does not contain nullables or could not contain
    # nullables, return X
    if !(T <: Nullable) && !(Nullable <: T)
        return X
    else
        # clean out nulls
        deleteat!(X, find(isnull, X))
        res = similar(X, eltype(T))
        # if the eltype is nullable, then safe to grab values without check
        if (T <: Nullable)
            for i in eachindex(X, res)
                @inbounds res[i] = X[i].value
            end
        # if eltype is Any, need to check each element
        else
            for i in eachindex(X, res)
                @inbounds res[i] = isa(X[i], Nullable) ? X[i].value : X[i]
            end
        end
        res
    end
end
```

And if we merge these I take it that we should merge the corresponding `dropnull` functions too? I split the function definitions here to reflect the split definitions there.